### PR TITLE
ko: Model name label on the Index page now translated by i18n

### PR DESCRIPTION
**Pull request by KOlofinmoyin**
Currently, the label of the "New resource" button is not being translated.

image

The natural way to translate this is probably to rely on display_resource_name. However this always displays the plural form of the name. In this PR we modify this method to accept an option to display the singular form where requested.
Additionally, we introduce the gem kaminary-i18n in the example app. If missing, Kaminari will be missing translations e.g. 'Next' and 'Last' buttons, and will break if we run the app in a different language. Visual testing of this PR requires this gem, so we consider it should be part of the change.

### CODE REVIEW

File: app/helpers/administrate/application_helper.rb [hunk 1]
```diff
@@ -43,10 +44,10 @@ def model_from_resource(resource_name)
       dashboard.try(:model) || resource_name.to_sym
     end
 
-    def display_resource_name(resource_name)
+    def display_resource_name(resource_name, opts = {})
       dashboard_from_resource(resource_name).resource_name(
-        count: PLURAL_MANY_COUNT,
-        default: default_resource_name(resource_name),
+        count: opts[:singular] ? SINGULAR_COUNT : PLURAL_MANY_COUNT,
+        default: default_resource_name(resource_name, !!opts[:singular]),
```
Comment (pablobm) [inline app/helpers/administrate/application_helper.rb:50]: In which situation is this used? I'm trying to reproduce it, but I can't get it to work :-/

I'm starting to think perhaps we should have specs for this, but I'm not sure how to best do it...

**Review by pablobm — CHANGES_REQUESTED** [review]
Thank you for the PR @KOlofinmoyin. You are on to something here, but I think it can be polished a bit.

Apart from the comments, could you please rebase the PR? There's a long list of commits stemming from other branches, and that could do with a cleanup.

File: app/helpers/administrate/application_helper.rb (hunk 1 above)
Comment (RaeRachael) [reply app/helpers/administrate/application_helper.rb:50]: this change is to recreate the behaviour of index.html.erb:47 as it was before the change to allow for translation, as such it is already tested in feature tests of the links generated on the adim/log_entry page of the example app.

Comment (RaeRachael) [reply app/helpers/administrate/application_helper.rb:50]: we can definitely look into making unit tests for this too though

File: Gemfile [hunk 2]
```diff
@@ -6,6 +6,7 @@ gem "administrate-field-image"
 gem "faker"
 gem "front_matter_parser"
 gem "globalid"
+gem "kaminari-i18n"
```
[the remark this answers is not in the document]
Comment (KOlofinmoyin) [reply Gemfile:9]: Cheers @pablobm done

File: app/helpers/administrate/application_helper.rb [hunk 3]
```diff
@@ -81,8 +82,8 @@ def clear_search_params
 
     private
 
-    def default_resource_name(resource_name)
-      resource_name.to_s.pluralize.gsub("/", "_").titleize
+    def default_resource_name(name, singular)
+      (singular ? name.to_s : name.to_s.pluralize).gsub("/", "_").titleize
```
[the remark this answers is not in the document]
Comment (KOlofinmoyin) [reply app/helpers/administrate/application_helper.rb]: Fixed 👍

File: app/helpers/administrate/application_helper.rb (hunk 1 above)
Comment (KOlofinmoyin) [reply app/helpers/administrate/application_helper.rb]: Done 👍

**Review by pablobm — CHANGES_REQUESTED** [review]
Almost there! I think if you add the `let` I suggest, this will be ready to go.

File: app/views/administrate/application/index.html.erb [hunk 4]
```diff
@@ -44,7 +44,7 @@ It renders the `_table` partial to display details about the resources.
     <%= link_to(
       t(
         "administrate.actions.new_resource",
-        name: page.resource_name.titleize.downcase
+        name: display_resource_name(page.resource_name, singular: true).downcase
```
[the remark this answers is not in the document]
Comment (RaeRachael) [reply app/views/administrate/application/index.html.erb:47]: That is very true, but I think it is a different PR task

File: spec/helpers/administrate/application_helper_spec.rb [hunk 5]
```diff
@@ -27,24 +33,32 @@
...
+        activerecord: {
+          models: {
+            customer: {
+              one: "User",
+              other: "Users",
             },
           },
-        }
+        },
+      }
```
[the remark this answers is not in the document]
Comment (KOlofinmoyin) [reply spec/helpers/administrate/application_helper_spec.rb:45]: Applied (done), thanks.

**Review by nickcharlton — APPROVED** [review]
I think this is good to go!

Could one of you open an issue for the `downcase` issue that was bought up?

**Review by pablobm — APPROVED** [review]
Good for me!

**Pull request merged**